codalab/lib/upload_manager.py

```diff
@@ -159,27 +127,23 @@ def _unpack_fileobj(self, source_filename, source_fileobj, dest_path, simplify_a
         if simplify_archive:
             self._simplify_archive(dest_path)
 
-    def _simplify_archive(self, path):
+    def _simplify_archive(self, path: str) -> None:
         """
         Modifies |path| in place: If |path| is a directory containing exactly
-        one file / directory that is not ignored, then replace |path| with that
-        file / directory.
+        one file / directory, then replace |path| with that file / directory.
         """
         if not os.path.isdir(path):
             return
 
-        files = [f for f in os.listdir(path) if not self._ignore_file_in_archive(f)]
+        files = os.listdir(path)
```

Comment:
I went ahead and removed this logic, as well. Although it's technically a breaking change in how CodaLab treats certain bundles, I don't think we want to keep it because the current behavior is problematic.

Current behavior:
- If you upload an archive to CodaLab with exactly two files: `README` and `.DS_Store`, CodaLab will just treat the bundle as a single file called `README`.
- If you upload an archive to CodaLab with exactly three files: `README`, `README2`, and `.DS_Store`, CodaLab will treat the bundle as a folder with three files: `README`, `README2`, and `.DS_Store`.

New behavior (with this PR):
- If you upload an archive to CodaLab with exactly two files: `README` and `.DS_Store`, CodaLab will treat the bundle as a folder with two files: `README` and `.DS_Store`.
- If you upload an archive to CodaLab with exactly three files: `README`, `README2`, and `.DS_Store`, CodaLab will treat the bundle as a folder with three files: `README`, `README2`, and `.DS_Store`.

For the two-file scenario, the current behavior is problematic because the user **has already included the .DS_Store file** in the fileobj they are uploading. Presumably, if they really wanted to ignore `.DS_Store`, they would have done so with `cl upload --exclude-patterns`, and the fileobj wouldn't have included `.DS_Store` in the first place. Because fileobj _does_ include both these files, the new behavior in this PR makes the most sense.

Moreover, note the inconsistency in the current behavior -- `.DS_Store` is removed from the final bundle _only in_ the 2-files case, but it is included in the 3-files case. This also shows why the current behavior isn't ideal.

I removed this logic now so I wouldn't have to replicate this kind of problematic / unnecessary logic for Blob Storage (thus simplifying blob storage implementation).

These scenarios are also covered in the `test_fileobj_single_tar_gz_with_dsstore_should_not_simplify_archive` and `test_fileobj_single_tar_gz_with_dsstore_should_not_simplify_archive_2` test cases.

Reply:
Yeah, this function is annoyingly complicated, so it's great to simplify.  I think what you're doing makes sense...I can't remember why we had local file paths.  However, this is a potentially disruptive change, so we should double check and test everything before deploying.

Reply:
Ah, I think I see why this might have been implemented in the first place. If you zip up a single file on a Mac from the Finder, it automatically adds a `__MACOSX` directory to the zip archive. I guess CodaLab wanted to ignore that directory so it could just treat such archives as a single file, rather than storing them as a bundle directory on CodaLab.

Nevertheless, the inconsistency I mentioned above still persists with `__MACOSX` directories, so I think we're still good to remove it.